control-plane/pkg/reconciler/channel/controller.go

```diff
@@ -89,10 +89,20 @@ func NewController(ctx context.Context, watcher configmap.Watcher, configs *conf
 		)
 	}
 
+	features := feature.FromContext(ctx)
+	caCerts, err := reconciler.getCaCerts()
+	if err != nil && (features.IsStrictTransportEncryption() || features.IsPermissiveTransportEncryption()) {
+		// We only need to warn here as the broker won't reconcile properly without the proper certs because the prober won't succeed
+		logger.Warn("Failed to get CA certs when at least one address uses TLS", zap.Error(err))
+	}
+
 	impl := kafkachannelreconciler.NewImpl(ctx, reconciler)
 	IPsLister := prober.IdentityIPsLister()
-	reconciler.Prober = prober.NewAsync(ctx, http.DefaultClient, "", IPsLister, impl.EnqueueKey)
 	reconciler.IngressHost = network.GetServiceHostname(configs.IngressName, configs.SystemNamespace)
+	reconciler.Prober, err = prober.NewComposite(ctx, configs.IngressPodPort, configs.IngressPodTlsPort, IPsLister, impl.EnqueueKey, &caCerts)
```

Comment:
@Rahul-Kumar-prog you need to keep the port string the same here:
```suggestion
	reconciler.Prober, err = prober.NewComposite(ctx, "", configs.IngressPodTlsPort, IPsLister, impl.EnqueueKey, &caCerts)
```
@pierDipi you changed the http port to `""` in https://github.com/knative-extensions/eventing-kafka-broker/pull/2164 - do you know if the TLS port should also be `""` on this PR?

Reply:
yes, the reason for this is that we probing the service of the Kafka channel which for http will use 80 and for https will use 443 which are both the defaults port